### PR TITLE
Fix flaky Cypress test cypress/e2e/widgets/stickers.spec.ts

### PR DESCRIPTION
**Pull request by andybalaam**
Fixes https://github.com/vector-im/element-web/issues/26017

2 change:

1. ~~Make sure we check for either of the two possible URLs we might see for a sticker. (Element Web first shows the thumbnail, but when that 404s it updates to a download link.)~~ Moved to https://github.com/matrix-org/matrix-react-sdk/pull/11471
2. Change the widely-used `viewRoomByName` utility to check for an exact match on the accessible name of a room when switching to it. This avoids us accidentally picking the wrong room if its name starts with the same string. The accessible name of a room can be unpredictable on some browsers/setups but it seems to work so far. The only other solution I can see would be to match on something other than the accessible name, which would go against the *-testing-library philosophy I think, but suggestions welcome.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: cypress/support/views.ts [hunk 1]
```diff
@@ -65,10 +65,7 @@ declare global {
 }
 
 Cypress.Commands.add("viewRoomByName", (name: string): Chainable<JQuery<HTMLElement>> => {
-    return cy
-        .findByRole("treeitem", { name: new RegExp("^" + name) })
-        .should("have.class", "mx_RoomTile")
-        .click();
+    return cy.findByRole("treeitem", { name }).should("have.class", "mx_RoomTile").click();
```
Comment (t3chguy) [inline cypress/support/views.ts:68]: ... Did you look at why its like this? Its described in the JSDoc.

image

File: cypress/e2e/widgets/stickers.spec.ts [hunk 2]
```diff
@@ -86,10 +86,12 @@ function expectTimelineSticker(roomId: string) {
...
     cy.get(".mx_EventTile_sticker > a").should("have.attr", "href").and("include", `/${roomId}/`);
 
-    // Make sure the image points at the sticker image
+    // Make sure the image points at the sticker image. We will briefly show it
+    // using the thumbnail URL, but as soon as that fails, we will switch to the
+    // download URL.
     cy.get<HTMLImageElement>(`img[alt="${STICKER_NAME}"]`)
         .should("have.attr", "src")
-        .and("match", /thumbnail\/somewhere\?/);
+        .and("match", /(thumbnail|download)\/somewhere/);
```
Comment (t3chguy) [inline cypress/e2e/widgets/stickers.spec.ts:94]: This seems like a bad idea. It means the test won't be consistently awaiting a ready state of the app. I think the issue is the use of `.and` - https://docs.cypress.io/guides/core-concepts/retry-ability we should not be chaining like this as it doesn't retry properly.

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: cypress/support/views.ts (hunk 1 above)
Comment (andybalaam) [reply cypress/support/views.ts:68]: Ah, I didn't spot that thanks. I'll take another look. I think we should check for it actually containing the exact text - this approach doesn't work for rooms with similar names.

Comment (andybalaam) [reply cypress/support/views.ts]: Another attempt in [a814f1d](https://github.com/matrix-org/matrix-react-sdk/pull/11440/commits/a814f1d66a0eafcc1d25b7c7368adcc000c9e986) - this finds the "Rooms" tree and then uses `findByText` inside there, which should find only the exact room name.

File: cypress/e2e/widgets/stickers.spec.ts (hunk 2 above)
Comment (andybalaam) [reply cypress/e2e/widgets/stickers.spec.ts]: [5ec4026](https://github.com/matrix-org/matrix-react-sdk/pull/11440/commits/5ec402666ddef21a220cdefe2b2d09d72690fc54) searches for the alt text and URL simultaneously, so the whole thing will be retried, and it looks for the download form only, meaning we retry if the thumbnail form is visible, so the UI is in a consistent state when this succeeds.

File: package.json [hunk 3]
```diff
@@ -62,7 +62,7 @@
         "@babel/runtime": "^7.12.5",
         "@matrix-org/analytics-events": "^0.6.0",
         "@matrix-org/matrix-wysiwyg": "^2.4.1",
-        "@matrix-org/react-sdk-module-api": "^1.0.0",
+        "@matrix-org/react-sdk-module-api": "^2.0.0",
```
Comment (t3chguy) [inline package.json:65]: ???

File: src/components/views/dialogs/ScrollableBaseModal.tsx
Comment (t3chguy) [inline src/components/views/dialogs/ScrollableBaseModal.tsx:1]: Ditto

File: src/modules/ProxiedModuleApi.ts
Comment (t3chguy) [inline src/modules/ProxiedModuleApi.ts:1]: Ditto

File: test/modules/ProxiedModuleApi-test.ts
Comment (t3chguy) [inline test/modules/ProxiedModuleApi-test.ts:1]: Er?

**Review by t3chguy — CHANGES_REQUESTED** [review]
You've picked up totally unrelated commits:

image

**Comment by andybalaam** [comment]
> You've picked up totally unrelated commits:
> 
> image

Sorry, no idea what happened. Rebased.

**Review by t3chguy — APPROVED** [review]

**Pull request merged**